### PR TITLE
fix: Treat Video compression 0 as false. 

### LINKED ISSUE #5191: When passing  "videoCompression":0, the output video does not work
### Current behavior:

In my `cypress.json` I have the following configuration:

```json
{
  "videoCompression":0
}
```

When running my specs with `cypress run` the videos that are generated of the tests seems broken somehow.

I'm trying to open them on a mac (mojave) with quicktime and vlc. Quicktime can't open the file, and vlc crashes when opening it.

### LINKED ISSUE #24595: Videos not viewable on Cypress Dashboard when videoCompression is set to 0
### Current behavior

I imagine that this is related to [this issue](https://github.com/cypress-io/cypress/issues/5191)

I've been trying to debug video frame skip issues when running Cypress via GitHub Actions ([issue here](https://github.com/cypress-io/cypress/issues/20468)). One thing that was suggested to me was to set the `videoCompression` to `0` in `cypress.config.ts`.

According to the GitHub Action's execution log, the upload was successful:

Screenshot 2022-11-08 at 16 49 20

However, when looking at [this run](ttps://dashboard.cypress.io/projects/iyj7ts/runs/75/) on the dashboard, the video is not watchable: 

Screenshot 2022-11-08 at 16 51 31

And there is this error in the console:
Screenshot 2022-11-08 at 16 51 12

### Desired behavior

_No response_

### Test code to reproduce

Relevant config items:

```js
{
  videoCompression: 0,
  viewportHeight: 800,
  viewportWidth: 1250,
}
```

### Cypress Version

10.11.0

### Node version

v16.18.0

### Operating System

ubuntu-20.04

### Debug Logs

```shell
I can email the run log if required, but would rather not share on here
```

### Other

_No response_

### PR DESCRIPTION
**Pull request by mjhenkes**
- Closes #5191
- Closes #24595

### Additional details

This Pull request also disables video compression for our tests in CI

The following table is a comparison between a develop ci run and a ci run in this branch, since it is not averaged over time the number just represent a snapshot in time and the real performance difference is probably somewhere in the middle.

|CI Job | [Before](https://app.circleci.com/pipelines/github/cypress-io/cypress/51530/workflows/092eb751-8b97-4655-988f-51991947f0f4) | [After](https://app.circleci.com/pipelines/github/cypress-io/cypress/51623/workflows/69f0314b-1596-4464-9262-4bcd584cc0a2) | Difference|
|---|---|---|---|
| driver-integration-tests-chrome | 10:26 | 9:32 |-0:54 |
| driver-integration-tests-chrome-beta | 10:25 | 9:10 | -1:15|
| driver-integration-tests-electron | 11:14 | 9:57 | -1:17 |
| driver-integration-tests-firefox | 8:48 | 10 | +1:12 |
| driver-integration-tests-webkit | 12:41 | 11:08 | -1:33 |
| run-app-component-tests-chrome | 4:06 | 4:02 | -0:04  |
| run-app-integration-tests-chrome | 9:22 | 9:22 |+/-0 |
| run-frontend-shared-component-tests-chrome | 5:01 | 3:37 | -1:24 |
| run-launchpad-component-tests-chrome | 2:20 | 2:34 | -0:14 |
| run-launchpad-integration-tests-chrome | 8 | 7:41 | -0:19 |
| run-reporter-component-tests-chrome | 1:55 | 2:07 | +0:12 |
| system-tests-chrome | 15:25 | 13:37 | -1:48 |
| system-tests-electron | 16:25 | 16:09 |-0:14 |
| system-tests-firefox | 11:28 | 12:05 |+0:37 |
| system-tests-webkit | 8:22 | 8: 09 |-0:13 |

Honeycomb metrics:
[Comparing average job duration between develop and this branch ](https://ui.honeycomb.io/cypress/environments/cypress-app/result/2cx1HXm2igd?hideMarkers)
[Average time spent in the video:post:procoessing span](https://ui.honeycomb.io/cypress/environments/cypress-app/result/vUGGtmzaD5i)

### Steps to test

* Set video compression to zero, watch the video.

### PR Tasks

- [ x ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by mjhenkes** [comment]
This still needs a Changlog entry, but i'll add that after the next release.

File: .circleci/workflows.yml [hunk 1]
```diff
@@ -493,7 +493,7 @@ commands:
               # internal PR
               CYPRESS_RECORD_KEY=$MAIN_RECORD_KEY \
               CYPRESS_INTERNAL_ENABLE_TELEMETRY="true" \
-              OTEL_RESOURCE_ATTRIBUTES="ci.branch=$CIRCLE_BRANCH,ci.job=$CIRCLE_JOB,ci.node-index=$CIRCLE_NODE_INDEX,ci.circle=$CIRCLECI,ci.build-url=$CIRCLE_BUILD_URL,ci.build-number=$CIRCLE_BUILD_NUM" \
+              OTEL_RESOURCE_ATTRIBUTES="ci.branch=$CIRCLE_BRANCH,ci.job=$CIRCLE_JOB,ci.node-index=$CIRCLE_NODE_INDEX,ci.circle=$CIRCLECI,ci.build-url=$CIRCLE_BUILD_URL,ci.build-number=$CIRCLE_BUILD_NUM,SHA1=$CIRCLE_SHA1,PR_NUMBER=$CIRCLE_PR_NUMBER" \
```
Comment (mjhenkes) [inline .circleci/workflows.yml]: adding to telemetry, i want to turn this into a custom detector, but thats for another pr

File: packages/server/lib/cypress.js [hunk 2]
```diff
@@ -32,7 +32,10 @@ const exit = async (code = 0) => {
     })
   }
 
-  telemetry.getSpan('cypress')?.end()
+  const span = telemetry.getSpan('cypress')
+
+  span?.setAttribute('exitCode', code)
```
Comment (mjhenkes) [inline packages/server/lib/cypress.js:37]: lets log out the exit code to know if jobs failed or not.

File: packages/server/lib/modes/run.ts [hunk 3]
```diff
@@ -321,9 +321,9 @@ async function postProcessRecording (options: { quiet: boolean, videoCompression
 
   // once this ended promises resolves
   // then begin processing the file
-  // dont process anything if videoCompress is off
+  // don't process anything if videoCompress is off
   // or we've been told not to upload the video
-  if (options.videoCompression === false || options.shouldUploadVideo === false) {
+  if (options.videoCompression === false || options.videoCompression === 0 || options.shouldUploadVideo === false) {
```
Comment (mjhenkes) [inline packages/server/lib/modes/run.ts:326]: This is the bug fix.

File: packages/server/lib/modes/run.ts [hunk 4]
```diff
@@ -676,7 +676,7 @@ async function waitForTestsToFinishRunning (options: { project: Project, screens
 
       span?.setAttributes({
         videoName,
-        videoCompression,
+        videoCompressionString: videoCompression.toString(),
```
Comment (mjhenkes) [inline packages/server/lib/modes/run.ts:679]: honeycomb decided this value was an int, so false === 0... i changed it to a string to get the real value.

File: system-tests/lib/system-tests.ts [hunk 5]
```diff
@@ -342,7 +342,7 @@ const isVideoSnapshotError = (err: Error) => {
   _.pull(added, sometimesAddedVideoSnapshotLine, sometimesAddedSpacingLine)
   _.pull(deleted, sometimesDeletedVideoSnapshotLine, sometimesAddedSpacingLine)
 
-  return _.isEqual(added, expectedAddedVideoSnapshotLines) && _.isEqual(deleted, expectedDeletedVideoSnapshotLines)
+  return _.isEqual(added, expectedAddedVideoSnapshotLines) && (deleted.length === 0 || _.isEqual(deleted, expectedDeletedVideoSnapshotLines))
```
Comment (mjhenkes) [inline system-tests/lib/system-tests.ts:345]: This is modifying an existing workaround for the firefox video error... previously when video was being compressed there were lines that were missing from the saved snapshot... since with compression off, the video name isn't logged, now there are not deleted lines.

File: system-tests/test/video_compression_spec.js [hunk 6]
```diff
@@ -101,3 +102,15 @@ describe('e2e video compression', () => {
     })
   })
 })
+
+describe('video compression 0', () => {
```
Comment (mjhenkes) [inline system-tests/test/video_compression_spec.js:106]: a new test for video compression 0

**Comment by mjhenkes** [comment]
I was honestly hoping to see better performance returns from system tests.

File: packages/frontend-shared/cypress.config.ts [hunk 7]
```diff
@@ -11,8 +11,8 @@ export default defineConfig({
   reporter: '../../node_modules/cypress-multi-reporters/index.js',
   reporterOptions: {
     configFile: '../../mocha-reporter-config.json',
-    videoCompression: false, // turn off video compression for CI
```
Comment (mschile) [inline packages/frontend-shared/cypress.config.ts:14]: Surprised this wasn't a typescript error.

Comment (mjhenkes) [reply packages/frontend-shared/cypress.config.ts:14]: same

File: system-tests/test/video_compression_spec.js [hunk 8]
```diff
@@ -39,11 +39,12 @@ describe('e2e video compression', () => {
   ].forEach((headed) => {
     systemTests.it(`passes (head${headed ? 'ed' : 'less'})`, {
       // videos are corrupted in firefox due to known issues
-      browser: '!firefox',
+      browser: ['chrome', 'electron'],
```
Comment (mschile) [inline system-tests/test/video_compression_spec.js]: Why was this changed? Was webkit also causing issues?

**Review by mschile — APPROVED** [review]

File: system-tests/test/video_compression_spec.js (hunk 8 above)
Comment (mjhenkes) [reply system-tests/test/video_compression_spec.js]: webkit was failing this test on my local machine with the error:

```
WebKit was launched, but the experimental feature was not enabled. Please add `experimentalWebKitSupport: true` to your config file to launch WebKit.
Error: WebKit was launched, but the experimental feature was not enabled. Please add `experimentalWebKitSupport: true` to your config file to launch WebKit.
```

So i assumed running the test against webkit was a waste 🤷‍♂️ 

Maybe it runs differently in CI?

Comment (mschile) [reply system-tests/test/video_compression_spec.js]: I see it being set [here](https://github.com/cypress-io/cypress/blob/c7da9f4ee61e3900af11a9e811934351ee5b0998/system-tests/lib/system-tests.ts#L934) but I haven't tried to run it locally. I assume it was working in CI before. Maybe just change it back?

File: packages/server/lib/modes/run.ts (hunk 3 above)
Comment (emilyrohrbough) [reply packages/server/lib/modes/run.ts:326]: `if (!options.videoCompress || !options.shouldUploadVideo)` ?

Comment (mjhenkes) [reply packages/server/lib/modes/run.ts:326]: i'd rather be explicit

**Review by emilyrohrbough — APPROVED** [review]
